pkg/actions/lua/databricks/client.go

```diff
@@ -0,0 +1,183 @@
+package databricks
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/url"
+	"regexp"
+	"strings"
+
+	"github.com/Shopify/go-lua"
+	"github.com/databricks/databricks-sdk-go"
+	"github.com/databricks/databricks-sdk-go/config"
+	"github.com/databricks/databricks-sdk-go/service/catalog"
+	"github.com/databricks/databricks-sdk-go/service/sql"
+)
+
+var r = regexp.MustCompile(`\\W`)
+
+var (
+	ErrInvalidTableName       = errors.New("invalid table name")
+	ErrInvalidTableNameLength = errors.New("invalid table name length")
+	ErrTableDeletionFailure   = errors.New("failed dropping an existing table")
+	ErrTableCreationFailure   = errors.New("failed creating table")
+)
+
+type Client struct {
+	workspaceClient *databricks.WorkspaceClient
+	ctx             context.Context
+}
+
+// https://docs.databricks.com/en/sql/language-manual/sql-ref-identifiers.html
+// https://docs.databricks.com/en/sql/language-manual/sql-ref-names.html
+func validateTableName(tableName string) error {
+	if r.Match([]byte(tableName)) {
+		return ErrInvalidTableName
+	} else if len(tableName) > 255 {
+		return ErrInvalidTableNameLength
+	}
+	return nil
+}
+
+func validateTableLocation(tableLocation string) error {
+	_, err := url.ParseRequestURI(tableLocation)
+	return err
+}
+
+func validateTableInput(tableName, location string) error {
+	errName := validateTableName(tableName)
+	errLocation := validateTableLocation(location)
+	return errors.Join(errName, errLocation)
```

Comment:
Why not just return the first error - expected result even when there is error in your query. It will report the first error it found and not error composed from all the invalid values.